### ISSUE TITLE
The execution is very slow, often times out

### ISSUE THREAD

**Issue by leonyah** [tarkal/highchart-lambda-export-server#3]
Thank you very much for this project!
I was able to get it to run, but it takes around 60+secs to generate a simple chart.
API gateway usually just times out, and half the time I see "0x03 error when performing chart generation" in the log.
Have you found a way to tune the Lambda environment to make this run quickly?
Thank you!
Leo

**Comment by tarkal** [comment]
Hi Leo,

I haven’t experienced that problem. Most of my charts render in 1-10 secs. Can you post a sample of the chart config you are using along with your lambda config and results.

Thx

Tarka

**Comment by leonyah** [comment]
That is so strange. I am trying to render a very basic chart:
{
	"title": {
		        "text": "Leo Chart"
	},
	"xAxis": {
		"categories": ["Jan", "Feb", "Mar", "Apr"]
	},
	"series": [
		        {
		            "type": "line",
		            "data": [1, 3, 2, 4]
		        },
		        {
		            "type": "line",
		            "data": [5, 3, 4, 2]
		        }
	]
}

I do get correct output occasionally, but most of the times it times out or throws the 0x03 error.

For the Lambda I am using your zip file as is.  Using Node.js 10.x is very slow (60+secs),  I also tried Node.js 12.x which seems to work faster (about 25secs) but also throws the 0x03 error occasionally.

**Comment by tarkal** [comment]
How much memory have you assigned to the function? It uses a reasonable amount so I would recommend assigning a minimum of 640 MB. Depending on the chart def I found I got performance increases up to about 1600 MB after which it seems to give diminishing returns. I did find that if I allocated less than 300 MB I got timeout and memory errors. I did I just created a brand new instance using the default package with 340MB and ran 100 tests using your chart def. I got 100% successful responses with an average response time of 874ms. Here is a screenshot of the longest invocation at 1529ms.



Let me know if increasing the memory helps.

Thx

Tarka
